From 4d51a097b9bfdfc95c0d40fb291197a4162f119e Mon Sep 17 00:00:00 2001
From: Brian Delwiche <delwiche@google.com>
Date: Sat, 24 Sep 2022 00:49:46 +0000
Subject: [PATCH] Add bounds check in avdt_scb_act.cc

Bug: 242535997
Test: BT unit tests, validated against researcher POC
Tag: #security
Ignore-AOSP-First: Security
Change-Id: Id3b665da0214e3c93f277e31077214fcd1d936e7
(cherry picked from commit 789d4bc617da23dc86d288c53c80a242d3a6850f)
Merged-In: Id3b665da0214e3c93f277e31077214fcd1d936e7
---
 system/stack/avdt/avdt_scb_act.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/system/stack/avdt/avdt_scb_act.cc b/system/stack/avdt/avdt_scb_act.cc
index 74c7cf264e..9a7c05d2a3 100644
--- a/system/stack/avdt/avdt_scb_act.cc
+++ b/system/stack/avdt/avdt_scb_act.cc
@@ -1023,6 +1023,11 @@ void avdt_scb_hdl_write_req(AvdtpScb* p_scb, tAVDT_SCB_EVT* p_data) {
 
   /* Build a media packet, and add an RTP header if required. */
   if (add_rtp_header) {
+    if (p_data->apiwrite.p_buf->offset < AVDT_MEDIA_HDR_SIZE) {
+      android_errorWriteWithInfoLog(0x534e4554, "242535997", -1, NULL, 0);
+      return;
+    }
+
     ssrc = avdt_scb_gen_ssrc(p_scb);
 
     p_data->apiwrite.p_buf->len += AVDT_MEDIA_HDR_SIZE;
-- 
2.39.0.rc1.256.g54fd8350bd-goog

